pkg/controllers/externalsecret/externalsecret_controller.go

```diff
@@ -299,6 +300,38 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 	}, nil
 }
 
+func createOrUpdate(ctx context.Context, c client.Client, obj client.Object, f func() error, fieldOwner string) error {
```

Comment:
This function is the same as `ctrl.CreateOrUpdate` but with the catch that we are using some of the client `CreateOptions` and `UpdateOptions` which are not available at `CreateOrUpdate` method